alchemiscale/storage/statestore.py

```diff
@@ -1710,14 +1710,50 @@ def set_tasks(
             transformation, Transformation, scope
         )
 
-    def set_task_priority(self, task: ScopedKey, priority: int):
-        q = f"""
-        MATCH (t:Task {{_scoped_key: "{task}"}})
-        SET t.priority = {priority}
-        RETURN t
-        """
+    def set_task_priority(self, task: Union[ScopedKey, List[ScopedKey]], priority: int):
+        if not priority >= 0:
+            raise ValueError("priority cannot be negative")
+
+        if isinstance(task, ScopedKey):
+            task = [task]
```

Comment:
> Just to simplify a bit and make it consistent with other methods, let's axe this method's ability to take a single task, since it avoids the asymmetry of input sometimes being a non-iterable but output always being an iterable.
> 
> We generally try to take this pattern at this lowest layer of the stack; it doesn't need to be user-friendly, but we aim for a fairly logical consistency where possible.
> 
> This will simplify the tests for these methods as well, reducing the input space.

Agreed. This will require changing compute tests using the single task inputs. I was trying to preserve previous functionality. Might be worthwhile doing a general consistency sweep before the next release.